frame/uniques/src/lib.rs

```diff
@@ -1409,4 +1448,30 @@ pub mod pallet {
 			Ok(())
 		}
 	}
+
+	impl<T: Config<I>, I: 'static> Pallet<T, I> {
+		fn is_collection_owner(
+			maybe_check_owner: &Option<T::AccountId>,
+			collection_details: &CollectionDetails<T::AccountId, DepositBalanceOf<T, I>>,
+		) -> DispatchResult {
+			if let Some(check_owner) = &maybe_check_owner {
+				ensure!(check_owner == &collection_details.owner, Error::<T, I>::NoPermission);
+			}
+
+			Ok(())
```

Comment:
I'm uneasy that `is_collection_owner(None, ...)` will return `Ok(())`. I can not be the owner and it will not throw a `NoPermission` error from what I can see?

Reply:
It is just shortcat function that, yea, bad reflect it own semantic, so i'll try come up with something better.

Reply:
I changed the names, ahah cache invalidation and naming, right?